ci-operator/step-registry/cucushift/chainupgrade/toversion/cucushift-chainupgrade-toversion-commands.sh

```diff
@@ -0,0 +1,732 @@
+        return 1
+    fi
+    export TARGET
+    return 0
+}
+
+function change_channel(){
+    local ret
+    oc adm upgrade channel --allow-explicit-channel "$1" || true
+    ret=$(oc get clusterversion version -o json|jq -r '.spec.channel')
+    if [[ "${ret}" != "$1" ]]; then
+        echo "Failed to update channel to $1, current channel is ${ret}!"
+        return 1
+    fi
+    return 0
+}
+
+if [[ -f "${SHARED_DIR}/kubeconfig" ]] ; then
+    export KUBECONFIG=${SHARED_DIR}/kubeconfig
+fi
+
+# Setup proxy if it's present in the shared dir
+if [[ -f "${SHARED_DIR}/proxy-conf.sh" ]]; then
+    # shellcheck disable=SC1091
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+# upgrade-edge file expects a comma separated releases list like target_release1,target_release2,...
+release_string="$(< "${SHARED_DIR}/upgrade-edge")"
+# shellcheck disable=SC2207
+TARGET_RELEASES=($(echo "$release_string" | tr ',' ' '))
+echo "Upgrade targets in ci config are ${TARGET_RELEASES[*]}"
+
+export OC="run_command_oc"
+
+mkdir -p /tmp/client
+export OC_DIR="/tmp/client"
+export PATH=${OC_DIR}:$PATH
+for target in "${TARGET_RELEASES[@]}"; do
+    # we need this DUMMY_TARGET_VERSION from ci config to do some pre-check
+    DUMMY_TARGET_VERSION="$(env "NO_PROXY=*" "no_proxy=*" oc adm release info "${target}" --output=json | jq -r '.metadata.version')"
+    TARGET_MAJOR_MINOR_VERSION="$(echo "${DUMMY_TARGET_VERSION}" | cut -d. -f1,2)"
+    TARGET_MINOR_VERSION="$(echo "${DUMMY_TARGET_VERSION}" | cut -d. -f2)"
+    export TARGET_MAJOR_MINOR_VERSION
+    export TARGET_MINOR_VERSION
+    
+    SOURCE_VERSION="$(oc get clusterversion --no-headers | awk '{print $2}')"
+    SOURCE_MINOR_VERSION="$(echo "${SOURCE_VERSION}" | cut -f2 -d.)"
+    export SOURCE_VERSION
+    export SOURCE_MINOR_VERSION
+
+    if ! get_target_from_available_update "${TARGET_MAJOR_MINOR_VERSION}"; then
+        echo "Fail to get target from available update for version: ${TARGET_MAJOR_MINOR_VERSION}"
+        exit 1
+    fi
+
+    echo -e "Source release version is: ${SOURCE_VERSION}; Source minor version is: ${SOURCE_MINOR_VERSION}"
+    echo -e "Target release version is: ${TARGET_VERSION}; Target minor version is: ${TARGET_MINOR_VERSION}"
+    # Target oc will be extract in the /tmp/client directory
+    extract_oc 
+
+    if check_ota_case_enabled "OCP-48683"; then
+        channel=$(oc get clusterversion version -o json|jq -r '.spec.channel')
+        
+        echo "Check updates to target_version in channel-last-hop is the same with channel-current-hop"
+        recommends_last_hop=$(oc get clusterversion version -o json|jq -r '.status.availableUpdates[]?.version'|grep "${TARGET_MAJOR_MINOR_VERSION}")
+        change_channel "candidate-${TARGET_MAJOR_MINOR_VERSION}"
+        recommends_current_hop=$(oc get clusterversion version -o json|jq -r '.status.availableUpdates[]?.version'|grep "${TARGET_MAJOR_MINOR_VERSION}")
```

Comment:
After `change_channel`, shall we need some wait to make the channel take effect ?